src/storage/src/cell_based_row_deserializer.rs

```diff
@@ -62,18 +61,16 @@ impl CellBasedRowDeserializer {
         let pk_vec_len = pk_with_cell_id.len();
         let cur_pk_bytes = &pk_with_cell_id[0..pk_vec_len - 4];
         let mut result = None;
-        if let Some(prev_pk_bytes) = &self.pk_bytes && prev_pk_bytes != cur_pk_bytes {
+        let cell_id_bytes = &pk_with_cell_id[pk_vec_len - 4..];
```

Comment:
Better to check length here. If key length < 4, return error.

Also, consider using `.split_at` function to create two slices.